src/components/MainContent-component/MainContent-Body/MainContentBodyLower/OneDayConatiner/TodoContainer/TodoItemUI.jsx

```diff
@@ -0,0 +1,102 @@
+import styled from 'styled-components';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faCheck, faTrashCan } from '@fortawesome/free-solid-svg-icons';
+import { useContext } from 'react';
+import { ThemeContext } from '../../../../../../context';
+
+const StyledTodoItemUI = styled.div`
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  height: 20%;
+  min-height: 20%;
+  padding-left: 20px;
+  font-size: 20px;
+  font-weight: 500;
+  margin: 0 5px;
+  border-radius: 10px;
+  margin-bottom: 5px;
+  &:hover {
+    background-color: #d5e9cb;
+  }
+`;
+
+const StyledTodoItemText = styled.p`
+  width: 65%;
+`;
+
+const StyledIconsContainer = styled.div`
+  display: flex;
+  align-items: center;
+  width: 35%;
+  justify-content: center;
+  column-gap: 13px;
+`;
+
+const StyledCheckIcon = styled(FontAwesomeIcon)`
+  &:hover {
+    cursor: pointer;
+  }
+`;
+const StyledTrashCanIcon = styled(FontAwesomeIcon)`
+  &:hover {
+    cursor: pointer;
+  }
+`;
+
+export default function TodoItemUI({ text }) {
+  const { selectedDayString, setSelectedDayTodoList, setSelectedDayDoneList } =
+    useContext(ThemeContext);
+
+  const localStorageTodoKey = `${selectedDayString}todo`;
+  const localStorageDoneKey = `${selectedDayString}done`;
+
+  function handleClickCheckIcon() {
+    const prevLocalStorageData = JSON.parse(
+      localStorage.getItem(localStorageTodoKey)
+    );
+    const prevItemLocalStorageIndex = prevLocalStorageData.indexOf(text);
+    prevLocalStorageData.splice(prevItemLocalStorageIndex, 1);
+    localStorage.setItem(
+      localStorageTodoKey,
+      JSON.stringify(prevLocalStorageData)
+    );
+
+    const prevLocalStorageDoneData =
+      localStorage.getItem(localStorageDoneKey) === null
+        ? []
+        : JSON.parse(localStorage.getItem(localStorageDoneKey));
+    prevLocalStorageDoneData.push(text);
+    localStorage.setItem(
+      localStorageDoneKey,
+      JSON.stringify(prevLocalStorageDoneData)
+    );
+
+    setSelectedDayTodoList(prevLocalStorageData);
+    setSelectedDayDoneList(prevLocalStorageDoneData);
+  }
+
+  // 로컬 스토리지에서 삭제하고 UI에서도 삭제를 해야함
+  function handleClickTrashIcon() {
+    const prevLocalStorageData = JSON.parse(
+      localStorage.getItem(localStorageTodoKey)
+    );
+
+    const prevLocalStorageIndex = prevLocalStorageData.indexOf(text);
+    prevLocalStorageData.splice(prevLocalStorageIndex, 1);
+    localStorage.setItem(
+      localStorageTodoKey,
+      JSON.stringify(prevLocalStorageData)
+    );
+    setSelectedDayTodoList(prevLocalStorageData);
+  }
```

Comment:
정확하진 않지만 이런느낌으로 리펙토링하면 코드 수를 줄일 수 있고, 로컬스토리지 접근도 간소화 할 수 있는 아이디어 일 것 같아요

```suggestion
// 로컬 스토리지 업데이트 로직을 별도의 함수로 분리
function updateLocalStorageItem(key, updateFunction) {
  const data = JSON.parse(localStorage.getItem(key)) || [];
  updateFunction(data);
  localStorage.setItem(key, JSON.stringify(data));
}

// 개선된 handleClickCheckIcon 함수
function handleClickCheckIcon() {
  updateLocalStorageItem(localStorageTodoKey, (data) => {
    const index = data.indexOf(text);
    if (index > -1) data.splice(index, 1);
  });
  updateLocalStorageItem(localStorageDoneKey, (data) => {
    data.push(text);
  });

  // 상태 업데이트 로직
}

// 개선된 handleClickTrashIcon 함수
function handleClickTrashIcon() {
  updateLocalStorageItem(localStorageTodoKey, (data) => {
    const index = data.indexOf(text);
    if (index > -1) data.splice(index, 1);
  });
```